### DISCUSSION TITLE
License for OnPremise installations

### DISCUSSION THREAD

**Discussion by sberer** [regulaforensics/DocumentReader-web-csharp-client#30] [Q&A]
We want to run the SDK calls against an OnPremise installation. A licence is already activated for the on-premise installation. Does this mean that a licence must also be sent with the call in the SDK or can this be omitted? 

```
var api = licenseFromEnv != null 
    ? new DocumentReaderApi(apiBaseUrl).WithLicense(licenseFromEnv)
    : new DocumentReaderApi(apiBaseUrl).WithLicense(licenseFromFile);

```

**Comment by esemenov1** [comment]
Hello!

In case you have an active license under the service you do not need to send it in the request (request).

In order to check the license status, you should send an HTTP query to the following endpoint http://localhost:8080/api/ping, which produces simple JSON output:
Example:
{
  "app-name": "Regula Document Reader Web API",
  "license-id": "00000000-0000-0000-0000-000000000000",
  "license-serial": "OL00000",
  "server-time": "2021-06-28 09:16:00.453891+00:00",
  "valid-until": "2022-12-31T00:00:00Z",
  "version": "5.7.128414.450"
}
